goth/assertions/monitor.py

```diff
@@ -279,3 +288,58 @@ def finished(self) -> bool:
         """Return True iif all assertions are done."""
 
         return all(a.done for a in self.assertions)
+
+
+class WaitableEventMonitor(EventMonitor[E]):
```

Comment:
Perhaps this should just be part of the base `EventMonitor` class? This would simplify the class hierarchy (and naming) without adding too much complexity to `EventMonitor`.

Reply:
This would simplify so many issues I had with naming things. Brilliant!